src/parsers.rs

```diff
@@ -204,8 +196,25 @@ fn time_second(i: &[u8]) -> IResult<&[u8], u32> {
     n_digit_in_range(i, 2, 0..=60)
 }
 
-fn time_millisecond(fraction: f32) -> u32 {
-    (1000.0 * fraction) as u32
+fn time_millisecond(i: &[u8]) -> IResult<&[u8], u32> {
+    let (i, mut digits) = take_while(is_digit)(i)?;
+    let mut l = digits.len();
+    if l > 3 {
+        digits = digits.get(0..3).unwrap();
+    }
+    let mut result = 0;
+    if l > 0 {
+        let digits = str::from_utf8(digits).unwrap(); // This can't panic, `digits` will only include digits.
+        result = digits.parse().unwrap();
+    }
+    while l < 3 {
+        result = result * 10;
+        l += 1;
+    }
+    //let fraction: f32 = format!("0.{}", digits).parse().unwrap(); // This can't panic, the string is a valid `f32`.
+
+    //let result = (1000.0 * fraction) as u32;
```

Comment:
```suggestion
```

Not necessary anymore.

Reply:
Sorry for leaving in leftover commented out code.